[NET10] CollectionView/CarouselView handers

davidortinau opened the pull request:
Updated

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/user-interface/controls/carouselview/index.md](https://github.com/dotnet/docs-maui/blob/e06352e71655d4479351c8a5b8bbd4e3d1ac722b/docs/user-interface/controls/carouselview/index.md) | [docs/user-interface/controls/carouselview/index](https://review.learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/carouselview/index?branch=pr-en-us-2995) |
| [docs/user-interface/controls/collectionview/index.md](https://github.com/dotnet/docs-maui/blob/e06352e71655d4479351c8a5b8bbd4e3d1ac722b/docs/user-interface/controls/collectionview/index.md) | [docs/user-interface/controls/collectionview/index](https://review.learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/collectionview/index?branch=pr-en-us-2995) |


<!-- PREVIEW-TABLE-END -->

davidortinau committed:
docs(.NET10): Note that optimized handlers are default for CollectionView/CarouselView on iOS/Mac Catalyst in .NET 10; update ms.date

jfversluis reviewed (changes requested):
I do think we want to add guidance here on how to opt-out?

jfversluis committed:
Document reverting to .NET 9 behavior for CarouselView

Added instructions to revert to .NET 9 behavior for CarouselView.

jfversluis committed:
Document reverting to .NET 9 handler for CarouselView

Added instructions to revert to .NET 9 behavior for CarouselView in MauiProgram.cs.

jfversluis committed:
Clarify handler recommendation for CollectionView

Updated the recommendation to use the new handler for CollectionView instead of CarouselView.
